### DISCUSSION TITLE
Configure light intensities in EV3

### DISCUSSION THREAD

**Discussion by techcatgato** [pybricks/support#1669] [Q&A]
Hi, i am using pybricks on an EV3. I want to create a color sensor calibration program, is there any function to configure the minimum and maximum light intensities of the EV3 color sensor?

**Comment by laurensvalk** [comment]
The intensity of the RGB LEDs can't be changed. I think they do already filter out ambient light, though.

If you mean to replicate the calibration of the older EV3 software, you could  store the lowest and highest values that occur in your application in a text file, and read from that the next time you run a program.

**Reply by techcatgato** [reply]
What i mean is functions that work like these blocks:
![image_2024-06-07_110307292](https://github.com/pybricks/support/assets/172017785/989fbef9-ed76-4334-8628-2083d1cf0261)
![image_2024-06-07_110321443](https://github.com/pybricks/support/assets/172017785/67caa7f9-d934-4cec-991d-6e66ac96bd03)

**Reply by laurensvalk** [reply]
Right. Those blocks will store a file with that number on the brick. You can do that too by storing a number in a text file. There isn't a ready-made oneliner for this though.

Also, please note those blocks don't actually make your sensor any more accurate. They just scale the measurement such that it goes from 0 to 100 between your minimum and maximum value.

**Reply by techcatgato** [reply]
> Right. Those blocks will store a file with that number on the brick. You can do that too by storing a number in a text file. There isn't a ready-made oneliner for this though.
> 
> Also, please note those blocks don't actually make your sensor any more accurate. They just scale the measurement such that it goes from 0 to 100 between your minimum and maximum value.

So theres no function to write or read such a file?

**Reply by laurensvalk** [reply]
You can write a file, just like normal Python.

**Reply by techcatgato** [reply]
> You can write a file, just like normal Python.

But how do i tell the sensor to use the file and in what format do i write it?

**Reply by laurensvalk** [reply]
Taking it one step at a time, can you write some code that will:

- write a number to a file, for example `123`.
- read a number from a file. It should give you the same number.

Taking it a step further, could you:
- write a function that takes a number and writes it to a file
- write a function that reads a number from a file.

The above is standard Python, unrelated to LEGO sensors, with numerous example online. This is a good way to learn :)

And then you essentially already have the code that corresponds to those "calibration blocks".

Then what remains is: Can you write some code that will:
- Stores the lowest value
- Stores the highest value

And:
- Reads the current value
- Scales it using the highest / lowest value.

**Reply by techcatgato** [reply]
I do know txt file management in python, I just don't know how to tell the sensors "Use the information from this file for min and max"

**Reply by laurensvalk** [reply]
If you're not sure what the calibration does, are you sure you want to use it in your code? Looking at it another way, what do you want to achieve?

You don't have to tell the sensors anything. The calibration blocks in the app don't do that either.  You'd just have to adapt the value that you read from the sensor. So something like:

```python
scaled_value = (measured_value - minimum_value) / (maximum_value - minimum_value) * 100
```

**Reply by techcatgato** [reply]
> If you're not sure what the calibration does, are you sure you want to use it in your code? Looking at it another way, what do you want to achieve?
> 
> You don't have to tell the sensors anything. The calibration blocks in the app don't do that either. You'd just have to adapt the value that you read from the sensor. So something like:
> 
> ```python
> scaled_value = (measured_value - minimum_value) / (maximum_value - minimum_value) * 100
> ```

Ohhh now i understand what you're saying. Sorry for the misunderstanding

**Reply by laurensvalk** [reply]
No worries! Now you'll remember it far longer as compared to getting a ready made solution right away :smile:
